Update gmail.rst

sjagr opened the pull request:
There are issues with the current docs as they don't address certain user configurations that prevent Symfony from emailing from their app. Mentioning two factor authentication and the less secure app setting is in my opinion, paramount.

Reference: http://stackoverflow.com/q/29085617/1188035

schuylr committed:
Update gmail.rst

There are issues with the current docs as they don't address certain user configurations that prevent Symfony from emailing from their app. Mentioning two factor authentication and the less secure app setting is in my opinion, paramount.

Reference: http://stackoverflow.com/q/29085617/1188035

WouterJ commented on cookbook/email/gmail.rst:
```diff
@@ -75,3 +75,9 @@ You're done!
+
+    Depending on your Gmail account settings, you may get authentication errors within your app.
+    You should ensure two-factor authentication and [also allow less secure apps to access your
+    account](https://support.google.com/accounts/answer/6010255)
```
this is markdown syntax, while this docs is writting in reStructured Text. Please update this PR to use that syntax:

```rst
.. note::

    Depending on your Gmail account settings, you may get authentication errors within your app.
    You should ensure two-factor authentication and 
    `also allow less secure apps to access your account`_.

.. _`also allow less secure apps to access your account`: https://support.google.com/accounts/answer/6010255
```

javiereguiluz commented:
@sjagr thanks for proposing this improvement!

I have a question for you regarding this phrase:

> You should ensure two-factor authentication [...]

I don't really understand what does it mean. Maybe we could change it for one of these alternatives:

  * `You should enable two-factor authentication [...]`
  * `You should ensure that two-factor authentication is enabled [...]`

schuylr committed:
Revisions to the gmail.rst doc

Revisions concerning the 2-Step-Verification and correct documentation syntax. Thanks @WouterJ and @javiereguiluz

schuylr committed:
Reflect links correctly in gmail.rst

sjagr commented:
Thanks @WouterJ for the format corrections.
@javiereguiluz Obviously a brain-fart on my part. I've made it much clearer and added a reference to Google Help for better understanding.

There can be an alternative suggestion of just turning 2-Step-Verification completely off, but I figured a Gmail user with that setting enabled (such as me) would want to leave it on.

javiereguiluz commented:
This pull request looks finished. If that's true, can we please remove the `In progress` label? Thanks.

xabbuh commented on cookbook/email/gmail.rst:
```diff
@@ -75,3 +75,13 @@ You're done!
+
+    Depending on your Gmail account settings, you may get authentication errors within your app.
+    If your Gmail account uses 2-Step-Verification, you should `generate an App password`_ to use for your
+    ``mailer_password`` parameter.
```
There is no need for this line break. Can you also please wrap lines after the first word that crosses the 72nd character instead?

javiereguiluz commented:
I'm finishing this PR in #5430.

@sjagr don't worry about not having finished your PR. This happens sometimes. I've taken your work and finished it in another PR. I've also reused all your original commits so you get full credit for your work. Thanks.

xabbuh commented:
Thank you for starting this @sjagr!
